internal/command/android.go

```diff
@@ -24,17 +24,16 @@ var (
 
 // Android build and package the fyne app for the android OS
 type Android struct {
-	Context []Context
+	CrossBuilder
 }
 
-// Name returns the one word command name
-func (cmd *Android) Name() string {
-	return "android"
-}
+var _ PlatformBuilder = (*Android)(nil)
+var _ Command = (*Android)(nil)
 
-// Description returns the command description
-func (cmd *Android) Description() string {
-	return "Build and package a fyne application for the android OS"
+func NewAndroidCommand() *Android {
+	r := &Android{CrossBuilder: CrossBuilder{name: "android", description: "Build and package a fyne application for the android OS"}}
+	r.builder = r
```

Comment:
The `CrossBuilder` is where the common code is and it is the interface called by the upper command. Still that `CrossBuilder` need to know details of what it is building, which is what the `PlatformBuilder` do. Provide the `CrossBuilder` logic the ability to build for a specific platform.

The other source of the design is that we are not using cli.App at the moment, but a `Command` structure. I think that there will be further reorganization here once we move to cli.App as part of preparing the move to fyne repository. Something I would like to push back to a later PR and so I have avoided so far touching much of the `Command` design for that reason.

Reply:
It seems like this `CrossBuilder` is a helper utility then, so it could be an internal detail and not part of the interface heirarchy. It could be a field within the platform builder and created as needed where the platform builder could be passed in - this removes the circular nature of the relationship.